### PR TITLE
feat(github): getUserInfo

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

GitHub connector's `getUserInfo`

https://www.notion.so/silverhand/Social-470168a733904b7ba2b1f52dd5b131d0

## Testing

UT

@logto-io/eng

### CODE REVIEW

**Review by gao-sun — APPROVED** [review]
lgtm

File: packages/core/src/connectors/github/index.ts [hunk 1]
```diff
@@ -57,3 +57,29 @@ export const getAccessToken: GetAccessToken = async (code) => {
...
+export const getUserInfo: GetUserInfo = async (accessToken: string) => {
+  const {
+    id,
+    avatar_url: avatar,
+    email,
+    name,
+  } = await got
+    .get(userInfoEndpoint, {
+      headers: {
+        authorization: `token ${accessToken}`,
```
Comment (simeng-li) [inline packages/core/src/connectors/github/index.ts:70]: Bearer?

File: packages/core/src/connectors/github/index.ts [hunk 2]
```diff
@@ -57,3 +57,29 @@ export const getAccessToken: GetAccessToken = async (code) => {
...
+      },
+    })
+    .json<{
+      id: number;
```
Comment (simeng-li) [inline packages/core/src/connectors/github/index.ts:74]: nit: prefer define a type, just make the code cleaner a little bit. 
```
type  ResponseType  = {}
.json<ResponseType>()
```

File: packages/core/src/connectors/github/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/connectors/github/index.ts:70]: It is `token`: https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#3-use-the-access-token-to-access-the-api

**Pull request merged**